tests/main/security-devpts/pts.exp

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/expect
+
+set timeout 2
+
+send "ls /dev/pts"
+expect "\[0-9\]*ptmx"
+
+# Launch app and checks contents of /dev/pts, should have only ptmx
+spawn sudo -i -u test /bin/sh -c /snap/bin/test-snapd-tools.sh
+expect "bash-4.3$"
+send "ls /dev/pts"
+expect "^ptmx"
+
+# From within confined app, open a pty
+send "python3"
+expect ">>>"
+send "import os, pty, sys"
+send "os.listdir('/dev/pts')"
+expect "\['ptmx'\]"
+send "pty.openpty()"
+send "os.listdir('/dev/pts')"
+expect "\['0', 'ptmx'\]"
+send "sys.exit(0)"
+expect "bash-4.3$"
+
+# From with confined app, verify pty was closed (assumes that the last program
+# closes it properly, which the above does)
+send "ls /dev/pts"
+expect "^ptmx"
```

Comment:
The above also doesn't work right. Here is a script that I think will do what you mean to do:

```
#!/usr/bin/expect

set timeout 2

spawn bash
send "ls /dev/pts\n"
expect "\[0-9]*ptmx" {} timeout { exit 1 }
send "exit\n"

# Launch app and checks contents of /dev/pts, should have only ptmx
spawn hello-world.sh
expect "bash-4.3\\$" {} timeout { exit 1 }
send "ls /dev/pts\n"
expect {
    timeout { exit 1 }
    "\[0-9]" { exit 1 }
    "ptmx"
}

# From within confined app, open a pty
send "python3\n"
expect ">>>" {} timeout { exit 1 }
send "import os, pty, sys\n"
send "os.listdir('/dev/pts')\n"
expect "\['ptmx'\]" {} timeout { exit 1 }
send "pty.openpty()\n"
send "os.listdir('/dev/pts')\n"
expect "\['0', 'ptmx'\]" {} timeout { exit 1 }
send "sys.exit(0)\n"
expect "bash-4.3\\$" {} timeout { exit 1 }

# From with confined app, verify pty was closed (assumes that the last program
# closes it properly, which the above does)
send "ls /dev/pts\n"
expect {
    timeout { exit 1 }
    "\[0-9]" { exit 1 }
    "ptmx"
}
send "exit\n"
```